src/java.base/share/classes/java/text/ListFormat.java

```diff
@@ -0,0 +1,557 @@
+     * {@code STANDARD} type, {@code FULL} style in US English:
+     * <table class="striped">
+     * <caption style="display:none">Standard/Full Patterns in US English</caption>
+     * <thead>
+     * <tr><th scope="col">Pattern Kind</th>
+     *     <th scope="col">Pattern String</th></tr>
+     * </thead>
+     * <tbody>
+     * <tr><th scope="row" style="text-align:left">start</th>
+     *     <td>"{0}, {1}"</td>
+     * <tr><th scope="row" style="text-align:left">middle</th>
+     *     <td>"{0}, {1}"</td>
+     * <tr><th scope="row" style="text-align:left">end</th>
+     *     <td>"{0}, and {1}"</td>
+     * <tr><th scope="row" style="text-align:left">2</th>
+     *     <td>"{0} and {1}"</td>
+     * <tr><th scope="row" style="text-align:left">3</th>
+     *     <td>""</td>
+     * </tbody>
+     * </table>
+     * Here are the resulting formatted strings with the above pattern array.
+     * <table class="striped">
+     * <caption style="display:none">Formatting examples</caption>
+     * <thead>
+     * <tr><th scope="col">Input String Array</th>
+     *     <th scope="col">Formatted String</th></tr>
+     * </thead>
+     * <tbody>
+     * <tr><th scope="row" style="text-align:left">["Foo", "Bar", "Baz", "Qux"]</th>
+     *     <td>"Foo, Bar, Baz, and Qux"</td>
+     * <tr><th scope="row" style="text-align:left">["Foo", "Bar", "Baz"]</th>
+     *     <td>"Foo, Bar, and Baz"</td>
+     * <tr><th scope="row" style="text-align:left">["Foo", "Bar"]</th>
+     *     <td>"Foo and Bar"</td>
+     * <tr><th scope="row" style="text-align:left">["Foo"]</th>
+     *     <td>"Foo"</td>
+     * </tbody>
+     * </table>
+     *
+     * @param patterns array of patterns, not null
+     * @throws IllegalArgumentException if the length {@code patterns} array is less than 5, or
+     *          any of {@code start}, {@code middle}, {@code end} patterns cannot be parsed.
+     * @throws NullPointerException if {@code patterns} is null.
+     */
+    public static ListFormat getInstance(String[] patterns) {
+        Objects.requireNonNull(patterns);
+        if (patterns.length != PATTERN_ARRAY_LENGTH) {
+            throw new IllegalArgumentException("Pattern array length should be " + PATTERN_ARRAY_LENGTH);
+        }
+        return new ListFormat(Locale.ROOT, Arrays.copyOf(patterns, PATTERN_ARRAY_LENGTH));
+    }
+
+    /**
+     * {@return the string that consists of the input strings, concatenated with the
+     * patterns specified, or derived from {@code Locale}, {@code Type}, and {@code Style}}
+     * @param input The list of input strings to format. There should at least
+     *              one String element in this list, otherwise an {@code IllegalArgumentException}
+     *              is thrown.
+     * @throws IllegalArgumentException if the length of {@code input} is zero.
+     * @throws NullPointerException if {@code input} is null.
+     */
+    public String format(List<String> input) {
```

Comment:
Shouldn't there be another format() method that takes in two Lists or a map, that way there is an option for the `FULL` strings and an option for the `SHORT`/`NARROW` strings.

Reply:
Not sure I understand the intention. The argument is a list of input elements to form the string. Can you please elaborate on your suggestion more?

Reply:
Sorry for the confusion, please disregard. I misinterpreted the List Pattern LDML specification, into thinking that the `STYLE`(s) not only affected the locale-specific delimiters but the elements themselves (which is not the case). That's why I thought a method that could take the different `STYLE` versions of the elements was needed.